tests/scripts/mbedtls_test.py

```diff
@@ -77,9 +77,7 @@ def __escaped_split(inp_str, split_char):
         """
         if len(split_char) > 1:
             raise ValueError('Expected split character. Found string!')
-        out = re.sub(r'(\\.)|' + split_char,
-                     lambda m: m.group(1) or '\n', inp_str,
-                     len(inp_str)).split('\n')
+        out = map(lambda x: re.sub(r'\\' + split_char, split_char,x), re.split(r'(?<!\\)' + split_char, inp_str))
         out = [x for x in out if x]
```

Comment:
I am happy with the new `map()` expression, but I was wondering: Is the line `out = [x for x in out if x]` necessary? It seems we are filtering out some test parameters that evaluate to false (for example `None`?) Why would that happen?